From 1d273c068bf7f7467ad30e177bbdd34f2ba14f2b Mon Sep 17 00:00:00 2001
From: Mattijs Korpershoek <mattijsx.korpershoek@intel.com>
Date: Tue, 1 Apr 2014 17:29:09 +0200
Subject: [PATCH 2/2] Add new target for libxml2 with "Schemas" feature

BZ: 184054

Adds a new target for building libxml with
schema validation support
This target is needed for our build system so
that we can see errors in the logs
when an xml is not respecting his xsd format.

Change-Id: Icdd854c9b020213c900ce2e153c210003f99393b
Category: aosp improvement
Domain: AOSP-Others
Origin: internal
Upstream-Candidate: no, proprietary
Signed-off-by: Mattijs Korpershoek <mattijsx.korpershoek@intel.com>
---
 Android.mk |   12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/Android.mk b/Android.mk
index 883f898..3f9a00e 100644
--- a/Android.mk
+++ b/Android.mk
@@ -97,3 +97,15 @@ LOCAL_CFLAGS += $(DISABLED_WARNING_FLAGS)
 LOCAL_SHARED_LIBRARIES += $(common_SHARED_LIBRARIES)
 LOCAL_MODULE:= libxml2
 include $(BUILD_HOST_STATIC_LIBRARY)
+
+
+# For the host on build systems with schema validation
+# ========================================================
+
+include $(CLEAR_VARS)
+LOCAL_SRC_FILES := $(common_SRC_FILES)
+LOCAL_C_INCLUDES := $(common_C_INCLUDES) external/icu4c/common
+LOCAL_SHARED_LIBRARIES := $(common_SHARED_LIBRARIES)
+LOCAL_CFLAGS := -DLIBXML_UNICODE_ENABLED -DLIBXML_REGEXP_ENABLED -DLIBXML_PATTERN_ENABLED -DLIBXML_SCHEMAS_ENABLED
+LOCAL_MODULE:= libxml2-schemas
+include $(BUILD_HOST_STATIC_LIBRARY)
-- 
1.7.10.4

